concepts/mesh_with_materials/python/material.py

```diff
@@ -17,35 +17,35 @@
 ASSETS_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
 
 
-def attach_billboard(stage, root, name="card"):
+def attach_billboard(stage: Usd.Stage, root: UsdGeom.Xform, name="card") -> UsdGeom.Mesh:
     billboard = UsdGeom.Mesh.Define(stage, str(root.GetPath()) + "/" + name)
     billboard.CreatePointsAttr(
         [(-430, -145, 0), (430, -145, 0), (430, 145, 0), (-430, 145, 0)]
     )
     billboard.CreateFaceVertexCountsAttr([4])
     billboard.CreateFaceVertexIndicesAttr([0, 1, 2, 3])
     billboard.CreateExtentAttr([(-430, -145, 0), (430, 145, 0)])
-    texCoords = billboard.CreatePrimvar(
+    texCoords = UsdGeom.PrimvarsAPI(billboard).CreatePrimvar(
         "st", Sdf.ValueTypeNames.TexCoord2fArray, UsdGeom.Tokens.varying
     )
     texCoords.Set([(0, 0), (1, 0), (1, 1), (0, 1)])
     return billboard
 
 
-def attach_surface_shader(stage, material, path):
+def attach_surface_shader(stage: Usd.Stage, material: UsdShade.Material, path: str) -> UsdShade.Shader:
     shader = UsdShade.Shader.Define(stage, path)
     shader.CreateIdAttr("UsdPreviewSurface")
     shader.CreateInput("roughness", Sdf.ValueTypeNames.Float).Set(0.4)
     shader.CreateInput("metallic", Sdf.ValueTypeNames.Float).Set(0.0)
 
-    material.CreateSurfaceOutput().ConnectToSource(shader, "surface")
+    material.CreateSurfaceOutput().ConnectToSource(shader.CreateInput("surface",  Sdf.ValueTypeNames.Color3f))
```

Comment:
not sure if this should be `CreateInput` or `CreateOutput`.  Also not sure if the type is correct.